packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,104 @@
+import {Component, Prop, h, Listen} from '@stencil/core';
+import {buildInteractiveResult, Result, SmartSnippet} from '@coveo/headless';
+import {LinkWithResultAnalytics} from '../result-link/result-link';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {ResultContextEvent} from '../result-template-components/result-template-decorators';
+
+@Component({
+  tag: 'atomic-smart-snippet-source',
+  shadow: false,
+})
+export class AtomicSmartSnippetSource implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Prop() source!: Result;
```

Comment:
We currently use a pattern where all props should have `reflect:true`. Additionally, I expect that some props like `source` and `smartSnippet` should be mutable.

Example:
```suggestion
  @Prop({reflect: true, mutable: true}) source!: Result;
```

Reply:
edit: I am wrong about `mutable`, I forgot that it only means that the prop is mutable from the inside. You don't need that option.